build_tools/scripts/integrate/bump_llvm.py

```diff
@@ -63,7 +63,7 @@ def main(args):
     iree_utils.git_create_branch(
         branch_name,
         checkout=True,
-        ref=f"{args.upstream_remote}/main",
+        ref=f"{args.upstream_remote}/{branch_name}",
```

Comment:
I think this should be fixed in other PR, while this PR is mainly for integrate.

Reply:
FWIW, I'm fine with including changes to the integrate tooling in integrates themselves. The changes are related enough, and I like to minimize the number of CI jobs needed (pre- and post- submit).

Reply:
I am also inclined to put that commit in this PR, since it is directly related to this integrate. It's also worthwhile to ensure that the upstream branch name doesn't conflict with an existing upstream branch (error out if it does).

Reply:
But since the CI is already running without the chagne now, we can push it in another commit

Reply:
@ScottTodd can you take a look at https://github.com/openxla/iree/pull/14463